docs/develop/typescript/cancellation.mdx

```diff
@@ -383,3 +383,20 @@ export async function shieldAwaitedInRootScope(): Promise<any> {
 ```
 
 <!--SNIPEND-->
+
+## Cancel an Activity from a Workflow {#cancel-an-activity}
+
+Canceling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout.
+If the Heartbeat is not invoked, the Activity cannot receive a cancellation request.
+When any non-immediate Activity is executed, the Activity Execution should send Heartbeats and set a [Heartbeat Timeout](/encyclopedia/detecting-activity-failures#heartbeat-timeout) to ensure that the server knows it is still working.
+
+When an Activity is canceled, an error is raised in the Activity at the next available opportunity.
```

Comment:
Maybe we can clarify this with something like:

"When an Activity is canceled, an error is thrown at the next opportunity, such as during a Heartbeat or when the Activity is about to make progress. This allows the Activity to detect and handle the cancellation."

Reply:
I only moved that subsection out from the Failure Detection page. I agree it needs rework, but if you don't mind, I would prefer to consider that one out of scope for this PR.

Reply:
NP!

Reply:
We'd be happy to help with rewording when you're comfortable with the technical content. Many people find it easier to separate those tasks (focus on tech, focus on org, focus on wording). It doesn't have to be multiple PRs. When you're happy, we can pull your branch, do a polishing pass in a subordinate branch, and allow you to review and merge that back into your PR before you submit your PR. This way quality tech coverage is aligned with our docs team.